tests/conference/test_conference_service.py

```diff
@@ -0,0 +1,88 @@
+
+
+def test_conference_create(session):
+    from dispatch.conference.service import create
+    from dispatch.conference.models import ConferenceCreate
+
+    resource_id = "000000"
+    resource_type = "resourcetype"
+    weblink = "https://www.example.com"
+    conference_id = "12345"
+    conference_challenge = "a0v0a0v9a"
+
+    conference_in = ConferenceCreate(
+        resource_id=resource_id,
+        resource_type=resource_type,
+        weblink=weblink,
+        conference_id=conference_id,
+        conference_challenge=conference_challenge,
+    )
+    conference = create(db_session=session, conference_in=conference_in)
+    assert conference
+    assert conference.resource_id == "000000"
+    assert conference.resource_type == "resourcetype"
+    assert conference.weblink == "https://www.example.com"
+    assert conference.conference_id == "12345"
+    assert conference.conference_challenge == "a0v0a0v9a"
+
+
+def test_conference_get(session, conference):
+    from dispatch.conference.service import get
+
+    test_conference = get(db_session=session, conference_id=conference.id)
+    assert test_conference.id == conference.id
+    assert test_conference.conference_challenge == conference.conference_challenge
+
+
+def test_conference_get_by_resource_id(session, conference):
+    from dispatch.conference.service import get_by_resource_id
+
+    test_conference = get_by_resource_id(db_session=session, resource_id=conference.resource_id)
+
+    assert test_conference.resource_id == conference.resource_id
+    assert test_conference.conference_challenge == conference.conference_challenge
+
+
+def test_conference_get_by_resource_type(session, conference):
+    """The service method returns a list of conferences that match a given resource type. We'll test
+    to ensure that the first and last items returned match the desired resource type."""
+    from dispatch.conference.service import get_by_resource_type
+
+    conferences = get_by_resource_type(db_session=session, resource_type=conference.resource_type)
+
+    assert conferences[0].resource_type == conference.resource_type
+    assert conferences[-1].resource_type == conference.resource_type
+
+
+def test_conference_get_by_conference_id(session, conference):
+    from dispatch.conference.service import get_by_conference_id
+
+    test_conference = get_by_conference_id(session, conference.conference_id)
+
+    assert test_conference.conference_id == conference.conference_id
+    assert test_conference.conference_challenge == conference.conference_challenge
+
+
+@pytest.mark.skip
+def test_conference_get_by_incident_id(session, conference):
+    from dispatch.conference.service import get_by_incident_id
+
+    """By the time we get here, several conferences have been added to the database. So this call to
+    get_by_incident_id will fail, as we assume only one() result there."""
+    test_conference = get_by_incident_id(db_session=session, incident_id=conference.incident_id)
```

Comment:
@mvilanova Here's this problem that I have. When running this test in isolation (`pytest -sv tests/conference/test_conference_service.py::test_conference_get_by_incident_id`), it succeeds, because in the [invocation](https://github.com/Netflix/dispatch/pull/267/files#diff-6147207fd13defcb5b55d0abe8ddf963R68), one, and only one, `conference` has been created. But when running the series of tests, the previous `conference`s defined earlier in the file have already been created. So the call to `one()` in `get_by_incident_id` will fail. Should I just move the test earlier in the file? Seems like the easy solution, but hides the problem. Tests should be idempotent and should work when run in isolation. They shouldn't depend on the results of the tests immediately before or after them.

Reply:
You should create a new incident for each invocation of `ConferenceFactory` you can either do this manually in your test or do it via `SubFactory`. There should never be more than one conference per incident. Also, I'm not sure where you are even associating an incident at all? Are you sure that the value isn't `None`? Quickly printing out the value of `conference` would indicate it hasn't been set.

```
Traceback (most recent call last):
  File "/home/kglisson/Workspace/dispatch/tests/conference/test_conference_service.py", line 73, in test_conference_get_by_incident_id
    assert conference.incident_id == "foo"
AssertionError: assert None == 'foo'
 +  where None = <dispatch.conference.models.Conference object at 0x7fe033779ba8>.incident_id
```

Reply:
Used a `SubFactory` to fix this in d13420f